pkg/cache/clusterqueue_test.go

```diff
@@ -223,58 +216,170 @@ func TestCohortCanFit(t *testing.T) {
 				},
 			},
 			wantFit: false,
-		},
-		"missing flavor": {
-			c: &Cohort{
-				Name: "C",
-				RequestableResources: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU:    5,
-						corev1.ResourceMemory: 5,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
+						"f2": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
 					},
-				},
-				Usage: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU:    5,
-						corev1.ResourceMemory: 5,
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    4,
+							corev1.ResourceMemory: 4,
+						},
+						"f2": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    4,
+							corev1.ResourceMemory: 4,
+						},
 					},
 				},
+				ResourceGroups: nil,
 			},
+		},
+		"missing flavor": {
 			request: FlavorResourceQuantities{
 				"f2": map[corev1.ResourceName]int64{
 					corev1.ResourceCPU:    1,
 					corev1.ResourceMemory: 1,
 				},
 			},
 			wantFit: false,
-		},
-		"missing resource": {
-			c: &Cohort{
-				Name: "C",
-				RequestableResources: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 5,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
 					},
-				},
-				Usage: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 3,
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
 					},
 				},
+				ResourceGroups: nil,
 			},
+		},
+		"missing resource": {
 			request: FlavorResourceQuantities{
 				"f1": map[corev1.ResourceName]int64{
 					corev1.ResourceCPU:    1,
 					corev1.ResourceMemory: 1,
 				},
 			},
 			wantFit: false,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 5,
+						},
+					},
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 3,
+						},
+					},
+				},
+				ResourceGroups: nil,
+			},
+		},
+		"nil lendingLimit can fit": {
```

Comment:
You should add two cases with same request for comparison:
1. lendingLimit disabled, can fit
2. lendingLimit enabled, can't fit.